src/Windows/Avalonia.Win32/Win32NativeToManagedMenuExporter.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Generic;
+using Avalonia.Controls;
+using Avalonia.Controls.Platform;
+
+#nullable enable
+
+namespace Avalonia.Win32
+{
+    internal class Win32NativeToManagedMenuExporter : INativeMenuExporter
+    {
+        private NativeMenu? _nativeMenu;
+
+        public void SetNativeMenu(NativeMenu? nativeMenu)
+        {
+            _nativeMenu = nativeMenu;
+        }
+
+        private IEnumerable<MenuItem> Populate (NativeMenu nativeMenu)
+        {
+            var items = new List<MenuItem>();
+
+            foreach (var menuItem in nativeMenu.Items)
+            {
+                if (menuItem is NativeMenuItemSeparator)
+                {
+                    items.Add(new MenuItem { Header = "-" });   
+                }
+                else if (menuItem is NativeMenuItem item)
+                {
+                    var newItem = new MenuItem { Header = item.Header, Icon = item.Icon, Command = item.Command, CommandParameter = item.CommandParameter };
+
+                    if(item.Menu != null)
+                    {
+                        newItem.Items = Populate(item.Menu);
+                    }
+                    else if (item.HasClickHandlers && item is INativeMenuItemExporterEventsImplBridge bridge)
+                    {
+                        newItem.Click += (s, e) => bridge.RaiseClicked();
+                    }
+
+                    items.Add(newItem);
+                }
+            }
+
+            return items;
+        }
+
+        public IEnumerable<MenuItem>? GetMenu ()
```

Comment:
Fix formatting in files also if it is possible. For example here you would usually write
```GetMenu()```
instead of
```GetMenu ()```

Reply:
done.